pkg/clock/clock.go

```diff
@@ -0,0 +1,166 @@
+package clock
+
+import (
+	"errors"
+	"sync"
+	"time"
+
+	clockapi "github.com/benbjohnson/clock"
+)
+
+var (
+	globalClock = clockapi.New()
+	mu          sync.Mutex
```

Comment:
I'm not positive if I need these -- definitely didn't want to use any locks on actual time function calls (Now, Since, Sleep...) that non-test production code would run so there wasn't any contention over time functions causing unneeded delays.

Reply:
To avoid racing you can use a sybc.RWMutex, so the read operations don't compete with each other but write operations have to get an exclusive lock